### PR TITLE
Fix the progress bar and add creator limitation

### PR DESCRIPTION
**Pull request by ChloePlanet**
### Changes
- [x] Added creator limitation (240) to avoid the `too many open files` error
Mac limitation:
```
$ launchctl limit maxfiles
	maxfiles    256            unlimited
```
- [x] Set logger level to Error to avoid warning messages flush.
- [x] Fixed progress bar flashing between `3/3` and `x/482`
- [x] Fixed progress bar ending with a unfinished status
```
449 / 452 [--------------------------------------------------------------------------------------------------------------------------------------------------------->_] 99.34% 7 p/s 1m7s
The generated files are output to folder `docs`
```
- [x] add docs

### CODE REVIEW

File: pkg/catalog/generator.go [hunk 1]
```diff
@@ -189,6 +189,9 @@ func (p *Generator) SetOptions(
 	return p
 }
 
+var maxCreatorCount = 240
```
Comment (cuminandpaprika) [inline pkg/catalog/generator.go]: Just curious, what made you decide to go with 240 here? I've tested it and it does seem to slow down quite a bit, so wondering how large we can make this.

Also, should this be a constant instead?

**Review by cuminandpaprika — COMMENTED** [review]
Looks good, just need to work out what value to set maxCreatorCount to, so that it doesn't trigger `too many files open` but also doesn't result in too much of a performance loss

File: pkg/catalog/generator.go (hunk 1 above)
Comment (ChloePlanet) [reply pkg/catalog/generator.go]: https://wilsonmar.github.io/maximum-limits/
image

I set it to 240 as the soft limit of my system is 256.

File: README.md [hunk 2]
```diff
@@ -83,8 +83,12 @@ services:
 
 ## How to use
 1. Set up environment
+
 `export SYSL_PLANTUML=http://www.plantuml.com/plantuml`
 
+On macOS, if your `launchctl limit maxfiles` output is too low (such as 256), or you encountered "too many open files" error, set the current session to higher limitation like:
+`sudo launchctl limit maxfiles 65536 200000`
```
Comment (cuminandpaprika) [inline README.md]:
```suggestion
On macOS, if your `launchctl limit maxfiles` setting is too low (e.g 256) you might see the error message "too many open files" when running make. 

You can set the current session limit higher with:
`sudo launchctl limit maxfiles 65536 200000`
```

**Review by cuminandpaprika — APPROVED** [review]
LGTM!

**Pull request merged**